src/doc/rustc/src/sanitizers.md

```diff
@@ -0,0 +1,96 @@
+| i686-unknown-linux-gnu      | address         |
+| x86_64-apple-darwin         | address, leak   |
+| x86_64-pc-windows-msvc      | address         |
+| x86_64-unknown-linux-gnu    | address, leak   |
+
+## AddressSanitizer
+
+AddressSanitizer (ASan) detects memory errors by instrumenting code at compile
+time and runtime to mark regions around allocated memory (i.e., red zones) as
+unaddressable (i.e., poisoned), quarantine and mark deallocated memory as
+unaddressable, and add checks before memory accesses. It uses a shadow memory
+mapping to store metadata information about whether a memory region is
+addressable. It can detect:
+
+* Heap-based buffer overflows, Stack-based buffer overflows, and other variants
+  of out-of-bounds reads and writes.
+* Use after free, double free, and other variants of expired pointer dereference
+  (also known as “dangling pointer”).
+* Initialization order bugs (such as [“Static Initialization Order
+  Fiasco”](https://en.cppreference.com/w/cpp/language/siof)).
+* Memory leaks.
+
+AddressSanitizer uses both instrumentation at compile time and runtime. It is
+recommended to recompile all code using AddressSanitizer for best results. If
+parts of the compiled code are not instrumented, AddressSanitizer may not detect
+certain memory errors or detect false positives.
+
+AddressSanitizer increases memory usage and also impacts performance due to the
+red zones and shadow memory mapping, and the added checks before memory
+accesses. AddressSanitizer and its runtime are not suitable for production use.
+
+For more information, see the [AddressSanitizer
+documentation](https://clang.llvm.org/docs/AddressSanitizer.html).
+
+## LeakSanitizer
+
+LeakSanitizer (LSan) detects memory leaks either as part of AddressSanitizer or
+as a standalone tool by instrumenting code at runtime to track all memory and
+thread management functions (i.e., interceptors), and searching for memory that
+remain allocated but are no longer reachable by any references in the program,
+at program termination or during specific checkpoints.
+
+LeakSanitizer can detect:
+
+* Memory allocated dynamically (e.g., via `malloc`, `new`) that are not freed or
+  deleted and is no longer referenced in the program (i.e., directly leaked
+  memory).
+* Memory allocated dynamically that is referenced by another memory that are not
+  freed or deleted and is no longer referenced in the program (i.e., indirectly
+  leaked memory).
+
+LeakSanitizer does not use instrumentation at compile time and works without
+recompiling all code using LeakSanitizer.
+
+LeakSanitizer impacts performance due to the interceptors and the checks for
+memory leaks at program termination. LeakSanitizer and its runtime are not
+suitable for production use.
+
+For more information, see the [LeakSanitizer
+documentation](https://clang.llvm.org/docs/LeakSanitizer.html).
+
+## Disclaimer
+
+The quality of the Sanitizers implementation and support varies across operating
```

Comment:
Do we want to say something about the stability of these sanitizers? For example, I assume we want to have the flexibility to ship a future version of Rust where they were removed?